src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/identifiers/CROIdentifier.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.digital.hmpps.personrecord.model.identifiers
+
+import org.apache.commons.lang3.builder.EqualsBuilder
+
+class CROIdentifier(inputCroId: String, inputFingerprint: Boolean, invalidInputCro: String = "") {
+
+  val croId: String = inputCroId
+  val fingerprint: Boolean = inputFingerprint
+
+  val valid: Boolean
+    get() = croId.isNotEmpty()
+
+  val invalidCro: String = invalidInputCro
+
+  override fun equals(other: Any?): Boolean {
+    return EqualsBuilder.reflectionEquals(this, other)
+  }
+
+  override fun hashCode(): Int {
+    return croId.hashCode()
```

Comment:
This needs to change to match the equals implementation. First answer on this link is good:

https://stackoverflow.com/questions/2265503/why-do-i-need-to-override-the-equals-and-hashcode-methods-in-java

```HashcodeBuilder.reflectionHashcode(this)```